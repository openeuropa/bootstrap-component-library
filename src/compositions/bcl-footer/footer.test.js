import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-data-footer/data";
import demoDataEU from "@openeuropa/bcl-data-footer/data-eu";
import demoDataEC from "@openeuropa/bcl-data-footer/data-ec";

const template = "@oe-bcl/bcl-footer/footer.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Footer", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});

describe("OE - Footer EU", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoDataEU)).resolves.toMatchSnapshot();
  });
});

describe("OE - Footer EC", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoDataEC)).resolves.toMatchSnapshot();
  });
});
