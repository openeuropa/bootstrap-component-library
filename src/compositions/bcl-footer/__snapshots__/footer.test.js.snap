// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Footer EC renders correctly 1`] = `
<jest>
  <footer
    class="bcl-footer bcl-footer--ec"
  >
    <div
      class="container"
    >
      <div
        class="pt-4 pt-lg-5 row"
      >
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold mb-2"
          >
            [Project name]
          </p>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque in sagittis dolor.
          </p>
        </div>
        <div
          class="col-12 col-lg-4 pb-4 pb-lg-0"
        >
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            Contact [project name]
          </p>
          <p>
            Contact information of the DG
          </p>
          <p
            class="fw-bold border-bottom pb-2 mt-3 mb-2"
          >
            Follow us on social media
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#facebook"
              />
            </svg>
            Facebook
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#twitter"
              />
            </svg>
            Twitter
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#linkedin"
              />
            </svg>
            Linkedin
          </a>
        </div>
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            About us
          </p>
          <p>
            Information about the DG
          </p>
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            Optional links
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
        </div>
      </div>
      <div
        class="pb-4 pb-lg-5 mt-4 mt-lg-5 bcl-footer__bordered-row row"
      >
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold pb-2 mb-2"
          >
            More information on:
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
        </div>
      </div>
      <div
        class="pb-4 pb-lg-5 bcl-footer__bordered-row row"
      >
        <div
          class="col-12 col-lg-4 pb-lg-4"
        >
          <p
            class="fw-bold pb-2 mb-2"
          >
            European Commission
          </p>
        </div>
        <div
          class="col-12 col-lg-4"
        >
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Commission
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Follow the EC on social media
            <svg
              class="ms-2 ms-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#box-arrow-up-right"
              />
            </svg>
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Resources for partners
          </a>
        </div>
        <div
          class="col-12 col-lg-4"
        >
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Language policy
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Cookies
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Privacy policy
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Legal notice
          </a>
        </div>
      </div>
    </div>
  </footer>
</jest>
`;

exports[`OE - Footer EU renders correctly 1`] = `
<jest>
  <footer
    class="bcl-footer bcl-footer--eu"
  >
    <div
      class="container"
    >
      <div
        class="pt-4 pt-lg-5 row"
      >
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold mb-2"
          >
            [Project name]
          </p>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque in sagittis dolor.
          </p>
        </div>
        <div
          class="col-12 col-lg-4 pb-4 pb-lg-0"
        >
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            Contact [project name]
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <p
            class="fw-bold border-bottom pb-2 mt-3 mb-2"
          >
            Follow us on social media
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#facebook"
              />
            </svg>
            Facebook
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#twitter"
              />
            </svg>
            Twitter
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#linkedin"
              />
            </svg>
            Linkedin
          </a>
        </div>
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            Optional links
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
        </div>
      </div>
      <div
        class="pb-4 pb-lg-5 mt-4 mt-lg-5 bcl-footer__bordered-row row"
      >
        <div
          class="col-12 col-lg-4 pb-4"
        >
          <img
            alt="logo"
            src="https://cdn1.fpfis.tech.ec.europa.eu/ecl/v3.0.2/eu/images/logo/standard-version/positive/logo-eu--en.svg"
          />
        </div>
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            Contact the EU
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Call us 00 800 6 7 8 9 10 11
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Use other telephone options
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Write us via our contact form
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Meet us at a local EU office
          </a>
          <p
            class="fw-bold border-bottom pb-2 pt-3 mb-2"
          >
            Social media
          </p>
          <p>
            Search for 
            <a
              class="mb-2"
              href="#"
            >
              EU social media channels
            </a>
          </p>
          <p
            class="fw-bold border-bottom pb-2 pt-3 mb-2"
          >
            Legal
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Language policy
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Privacy policy
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Legal notice
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Cookies
          </a>
        </div>
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold border-bottom pb-2 pt-4 pt-lg-0 mb-2"
          >
            EU institutions
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Parliament
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Council
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Council of the European Union
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Commission
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Court of Justice of the European Union
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Central Bank
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Court of Auditors
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European External Action Service
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Economic and Social Committee
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Committee of the Region
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Investment Bank
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Ombudsman
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Data Protection Supervisor
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Data Protection Board
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            European Personnel Selection Office
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Publications Office of the European Union
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Agencies
          </a>
        </div>
      </div>
    </div>
  </footer>
</jest>
`;

exports[`OE - Footer renders correctly 1`] = `
<jest>
  <footer
    class="bcl-footer bcl-footer--neutral"
  >
    <div
      class="container"
    >
      <div
        class="pt-4 pt-lg-5 row"
      >
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold mb-2"
          >
            [Project name]
          </p>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque in sagittis dolor.
          </p>
        </div>
        <div
          class="col-12 col-lg-4 pb-4 pb-lg-0"
        >
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            Contact [project name]
          </p>
          <p>
            Contact information of the DG
          </p>
          <p
            class="fw-bold border-bottom pb-2 mt-3 mb-2"
          >
            Follow us on social media
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#facebook"
              />
            </svg>
            Facebook
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#twitter"
              />
            </svg>
            Twitter
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            <svg
              class="me-2-5 bi icon--xs"
            >
              <use
                xlink:href="/icons.svg#linkedin"
              />
            </svg>
            Linkedin
          </a>
        </div>
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            About us
          </p>
          <p>
            Information about the DG
          </p>
          <p
            class="fw-bold border-bottom pb-2 mb-2"
          >
            Optional links
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
        </div>
      </div>
      <div
        class="pb-4 pb-lg-5 mt-4 mt-lg-5 bcl-footer__bordered-row row"
      >
        <div
          class="col-12 col-lg-4"
        >
          <p
            class="fw-bold pb-2 mb-2"
          >
            More information on:
          </p>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
          <a
            class="d-block standalone mb-1"
            href="#"
          >
            Link
          </a>
        </div>
      </div>
    </div>
  </footer>
</jest>
`;
